File: gui_view/src/main/scala/com/goexp/galgame/gui/util/res/Local.scala
package com.goexp.galgame.gui.util.res

import com.goexp.galgame.gui.util.cache.{AppCache, Cache}
import javafx.scene.image.Image

object Local {
  def getLocal(name: String): Image = getLocal(name, AppCache.imageMemCache)

  private def getLocal(name: String, imageMemCache: Cache[String, Image]) =
    imageMemCache.get(name) match {
      case Some(img) => img
      case None =>
        val image = new Image(this.getClass.getResource(name).toExternalForm)
        imageMemCache.put(name, image)
        image
    }
}

File: gui_view/src/main/scala/com/goexp/galgame/gui/util/res/Local.java
package com.goexp.galgame.gui.util.res;

import com.goexp.galgame.gui.util.cache.AppCache;
import com.goexp.galgame.gui.util.cache.Cache;
import javafx.scene.image.Image;

public class Local {

    public static Image getLocal(String name) {
        return getLocal(name, AppCache.imageMemCache());
    }

    private static Image getLocal(String name, Cache<String, Image> imageMemCache) {
        return imageMemCache.get(name).orElseGet(() -> {
            final var image = new Image(Local.class.getResource(name).toExternalForm());

            imageMemCache.put(name, image);
            return image;

        });
    }
}
